### DISCUSSION TITLE
Convert: Show on the last line of the Progress

### DISCUSSION THREAD

**Discussion by hollysheep** [ImageMagick/ImageMagick#7053] [Help]
### ImageMagick version

6.9.11

### Operating system, version and so on

Ubuntu 22.04

### Description

I use a custom action in thunar file manager to start converting images to jpg. The script starts a terminal window where i can see the progress. When using Image Magick i use the monitor option. But the output is hard to read. Its something like:

```
Save/Image//media/data/image-1
Save/Image//media/data/image-1
...
Save/Image//media/data/image-1
Save/Image//media/data/image-1
Save/Image//media/data/image-1
Save/Image//media/data/image-1697110196315330492/neu[002-new.jpg]: 3675 of 3676, 100% complete
```

The only interesting part is the last line because i see what file is converted at the moment. Can i output only the last line and skip the rest? i don't need it.

**Comment by urban-warrior** [comment]
You can use `grep` to only extract select monitoring output.  Or you can remove the monitoring and use `-verbose` instead which only outputs one info line for each image read or written.

**Comment by hollysheep** [comment]
-verbose works good, now i have 2 lines. I guess the first line gives information about the input file and the second is the convertion?

grep doesn't work here. The output is:

```
input.jpg JPEG 2500x3544 2500x3544+0+0 8-bit sRGB 789445B 0.090u 0:00.091
input.jpg=>output.jpg JPEG 2500x3544 2500x3544+0+0 8-bit sRGB 771185B 0.080u 0:00.085
```

`convert .... | grep "=>"`

should give my the second line only - but it gives my both. I looks like grep isn't working here for some reasons.

**Comment by hollysheep** [comment]
When i try

`convert input output > text-file`

i don't have any output in the terminal but an empty text file. With

`convert input output 2> text-file`

the normal output is written in the text file. But 2> is the stderr (error) and not stdout. But there is no error. The image is converted correctly. Why is the output error and not standard out?

**Reply by urban-warrior** [reply]
Suppose you want to create a pipeline.  If we write verbose or monitoring information to stdout, the monitoring would step on the image output.  For example,

```
$ magick -verbose rose: pnm:- | magick identify -
PNM:ROSE=>ROSE PNM 70x46 70x46+0+0 8-bit sRGB 9673B 0.000u 0:00.000
rose:=>ROSE PNM 70x46 70x46+0+0 8-bit sRGB 9673B 0.000u 0:00.000
-=>/tmp/magick-rntb7_FstYe-1NfUzWITPQAedeXE8i3o PPM 70x46 70x46+0+0 8-bit sRGB 9729B 0.000u 0:00.000
$ magick -verbose rose: pnm:- |& magick identify -
identify: no decode delegate for this image format `' @ error/constitute.c/ReadImage/746.
```

**Comment by snibgo** [comment]
> -verbose works good, now i have 2 lines. I guess the first line gives information about the input file and the second is the convertion?

The first line is the input, the second line is the output. Insert `-verbose` just before the output to get only that, like this:
```
magick in.png -verbose out.png
```
You can also use `+verbose` to turn verbosity off. For example:
```
magick -verbose in.png +verbose out.png
```
This gives verbosity for the input only.
